src/AssetsPage/AssetsTable/AssetsTable.test.jsx

```diff
@@ -24,31 +28,335 @@ const defaultProps = {
+        button.matchesElement(<button>Yes, delete.</button>));
+
+      trashButtons.at(0).simulate('click');
+      deleteButton.at(0).simulate('click');
+
+      expect(deleteAssetSpy).toHaveBeenCalledTimes(1);
+      expect(deleteAssetSpy).toHaveBeenCalledWith(
+        defaultProps.assetsParameters,
+        defaultProps.assetsList[0].id,
+      );
+    });
+    it('closes on deleteAsset call', () => {
+      wrapper = mount(
+        <AssetsTable
+          {...defaultProps}
+        />,
+      );
+
+      const deleteButtons = wrapper.find('button');
+
+      const trashButtons = deleteButtons.filterWhere(button => button.hasClass('fa-trash'));
+      const modal = wrapper.find('[role="dialog"]');
+
+      const deleteButton = deleteButtons.filterWhere(button =>
+        button.matchesElement(<button>Yes, delete.</button>));
+
+      trashButtons.at(0).simulate('click');
+
+      deleteButton.at(0).simulate('click');
+
+      expect(modal.hasClass('modal-open')).toEqual(false);
+      expect(wrapper.state('modalOpen')).toEqual(false);
+    });
+  });
+  describe('onDeleteClick', () => {
+    let trashButtons;
+
+    beforeEach(() => {
+      wrapper = mount(
+        <AssetsTable
+          {...defaultProps}
+        />,
+      );
+
+      trashButtons = wrapper.find('button').filterWhere(button => button.hasClass('fa-trash'));
+    });
+    it('sets modalOpen to true', () => {
+      trashButtons.at(0).simulate('click');
+      expect(wrapper.state('modalOpen')).toEqual(true);
+    });
+    it('sets assetToDelete correctly', () => {
+      const trashButtonAriaLabel = `Delete ${defaultProps.assetsList[0].display_name}`;
+      expect(trashButtons.at(0).prop('aria-label')).toEqual(trashButtonAriaLabel);
+
+      trashButtons.at(0).simulate('click');
+      expect(wrapper.state('assetToDelete')).toEqual(defaultProps.assetsList[0]);
+    });
+    it('sets elementToFocusOnModalClose correctly', () => {
+      trashButtons.at(0).simulate('click');
+
+      expect(trashButtons.at(0).matchesElement(wrapper.state('elementToFocusOnModalClose'))).toEqual(true);
+    });
+  });
+  describe('focus', () => {
+    let closeButton;
+    let deleteButtons;
+    let modal;
+    let trashButtons;
+    let mockDeleteAsset;
+
+    beforeEach(() => {
+      wrapper = mount(
+        <AssetsTable
+          {...defaultProps}
+        />,
+      );
+
+      deleteButtons = wrapper.find('button');
+      trashButtons = deleteButtons.find('button').filterWhere(button => button.hasClass('fa-trash'));
+
+      modal = wrapper.find('[role="dialog"]');
+      closeButton = modal.find('button').filterWhere(button => button.matchesElement(<button><span>&times;</span></button>));
+    });
+
+    it('moves from modal to trashcan on modal close', () => {
+      trashButtons.at(0).simulate('click');
+
+      expect(closeButton.at(0).matchesElement(document.activeElement)).toEqual(true);
+
+      closeButton.at(0).simulate('click');
```

Comment:
Is the `at(0)` necessary if you're filtering on the buttons within the modal specifically? There should only be one close button from that group of buttons.  Or is the `at(0)` supposed to be an extra validation?

Reply:
I checked, and it doesn't appear to be necessary. I think I got in the habit of doing that for instances where the ReactWrapper had more than one node in it, in which case it would complain with something like ```Method “simulate” is only meant to be run on a single node. 4 found instead.```. I think it might actually be better to leave it out, since it allows for the test to fail if more than the expected number of nodes are in the wrapper. If we find more than one closeButton, that's probably a problem, and doing closeButton.at(0) would mask that.